src/ccpp_scheme_utils.F90

```diff
@@ -0,0 +1,101 @@
+module ccpp_scheme_utils
+
+   ! Module of utilities available to CCPP schemes
+
+   use ccpp_constituent_prop_mod, only: ccpp_model_constituents_t, int_unassigned
+
+   implicit none
+   private
+
+   !! Public interfaces
+   public :: ccpp_initialize_constituent_ptr ! Used by framework to initialize
+   public :: ccpp_constituent_index          ! Lookup index constituent by name
+   public :: ccpp_constituent_indices        ! Lookup indices of consitutents by name
+
+   !! Private module variables & interfaces
+
+   ! initialized set to .true. once hash table pointer is initialized
+   logical                                  :: initialized = .false.
+   type(ccpp_model_constituents_t), pointer :: constituent_obj => NULL()
+
+   private :: uninitialized
+
+contains
+
+   subroutine uninitialized(caller, errcode, errmsg)
```

Comment:
I am a little confused by the name of this subroutine. Since the routine returns a failure when the data is NOT initialized, this routine should be called `initialized`. 

Q: Check if this code is initialized?
A: Error, it is not initialized.

Reply:
In the module, the routine is called when we already know the module has not been initialized. It would be a lot more efficient as a logical function but one with side effects (setting `errmsg` and `errcode`). Is that acceptable coding here?

Reply:
In this case it's fine with me, thanks for the explanation

Reply:
I thought of a clearer way to check the initialization. Also, I realized that I was not always initializing the error variables if present. Please let me know if my latest change is _not_ an improvement in your eyes.

Reply:
I like the new solution better. While reviewing it, I thought of another thing. The file is called ccpp_scheme_utils, but it looks like these tools specific to the constituents. Should the file be renamed accordingly?